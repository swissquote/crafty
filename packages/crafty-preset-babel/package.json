{
  "name": "@swissquote/crafty-preset-babel",
  "version": "1.13.5",
  "license": "Apache-2.0",
  "repository": "https://github.com/swissquote/crafty.git",
  "publishConfig": {
    "access": "public"
  },
  "main": "src/index.js",
  "dependencies": {
    "@babel/core": "7.12.3",
    "@babel/runtime": "7.12.1",
    "@rollup/plugin-babel": "5.2.1",
    "@swissquote/babel-preset-swissquote": "1.13.5",
    "@swissquote/crafty-preset-eslint": "1.13.5",
    "@swissquote/crafty-preset-terser": "1.13.5",
    "babel-loader": "8.1.0",
    "babel-plugin-istanbul": "6.0.0",
    "babel-preset-jest": "26.5.0",
    "gulp-babel": "8.0.0",
    "gulp-concat": "2.6.1",
    "gulp-eslint7": "0.3.0",
    "gulp-newer": "1.4.0",
    "gulp-sourcemaps": "2.6.5",
    "gulp-terser": "1.4.0"
  },
  "engines": {
    "node": ">=10.0.0"
  }
}
