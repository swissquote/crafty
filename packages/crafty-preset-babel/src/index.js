const path = require("path");

const MODULES = path.join(__dirname, "..", "node_modules");

module.exports = {
  presets: [
    require.resolve("@swissquote/crafty-preset-eslint"),
    require.resolve("@swissquote/crafty-preset-terser")
  ],
  defaultConfig(config) {
    return {
      bundleTypes: { js: "js" }
    };
  },
  jest(crafty, options) {
    options.moduleDirectories.push(MODULES);
    options.transform["\\.(js|jsx)$"] = require.resolve("./jest-transformer");
    options.moduleFileExtensions.push("jsx");

    const babelConfigurator = require("@swissquote/babel-preset-swissquote/configurator");
    options.globals.BABEL_OPTIONS = babelConfigurator(
      crafty,
      {},
      { environment: "test" }
    );
  },
  bundleCreator(crafty) {
    const configurators = { js: {} };

    if (
      crafty.config.loadedPresets.some(
        preset => preset.presetName === "@swissquote/crafty-runner-gulp"
      )
    ) {
      configurators.js["gulp/babel"] = (
        _crafty,
        bundle,
        gulp,
        StreamHandler
      ) => {
        const createTask = require("./gulp");
        gulp.task(bundle.taskName, createTask(_crafty, bundle, StreamHandler));
        _crafty.watcher.add(bundle.watch || bundle.source, bundle.taskName);
      };
    }

    return configurators;
  },
  rollup(crafty, bundle, rollupConfig) {
    const babelConfigurator = require("@swissquote/babel-preset-swissquote/configurator-rollup");
    const { hasRuntime, options } = babelConfigurator(crafty, bundle);

    if (hasRuntime) {
      rollupConfig.input.external.push(/@babel\/runtime/);
    }

    rollupConfig.input.plugins.babel = {
      plugin: require("@rollup/plugin-babel"),
      weight: 20,
      options
    };
  },
  webpack(crafty, bundle, chain) {
    chain.resolve.extensions.add(".jsx");
    chain.resolve.modules.add(MODULES);
    chain.resolveLoader.modules.add(MODULES);

    const babelConfigurator = require("@swissquote/babel-preset-swissquote/configurator-webpack");
    const options = babelConfigurator(crafty, bundle);

    // EcmaScript 2015+
    chain.module
      .rule("babel")
      .test(/\.jsx?$/)
      .exclude.add(/(node_modules|bower_components)/)
      .end()
      .use("babel")
      .loader(require.resolve("babel-loader"))
      .options(options);
  }
};
