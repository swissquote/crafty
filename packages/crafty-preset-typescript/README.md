<table>
<tr><th>Compatible Runners</th><td>

- [Webpack](05_Packages/02_crafty-runner-webpack.md)

</td></tr>
<tr><th>Test Runners</th><td>

- [Jest](05_Packages/05_crafty-preset-jest.md)

</td></tr>
<tr><th>Linters</th><td>

Provides ESLint, with full configuration for TypeScript. Based on the same configuration we use in JavaScript.

</td></tr>
</table>

[TOC]

## Description

Microsoft presents TypeScript as "JavaScript that scales".
TypeScript is a superset of JavaScript that allows you to type your code.
Not everything needs to have a type as the tool is powerful enough to infer most of the types.

TypeScript has types but it also supports the ESNext Specification.

## Features

TypeScript being a superset of JavaScript, features supported by [our Babel preset](../05_crafty-preset-babel/JavaScript_Features.md) apply here.

Or you can check out the [TypeScript specific features](TypeScript_Features.md)

## Linting

Like any language, TypeScript has best practices and practices that you shouldn't do.
With the help of ESLint we check the code for common mistakes and formatting errors.

[Read more](../05_crafty-preset-eslint/TypeScript_Linting.md)

## Installation

```bash
npm install @swissquote/crafty-preset-typescript --save
```

```javascript
module.exports = {
  presets: [
    "@swissquote/crafty-preset-typescript",
    "@swissquote/crafty-runner-webpack", // optional
  ],
  js: {
    app: {
      runner: "webpack", // webpack (optional if you have a single runner defined)
      source: "js/app.ts",
    },
  },
};
```

## Usage with Webpack

You have to configure TypeScript with the `tsconfig.json` file that you put at the root of the project.

Some options can't be auto-configured through the loaders / plugins so that's why you have to do it yourself.
Moreover, the integration with your IDE is simplified if you have a complete `tsconfig.json` in your project.

## Usage with Jest

If you include both `crafty-preset-typescript` and `crafty-preset-jest`.
When running your tests with `crafty test` this preset will be use to convert all `.ts`, `.tsx`, `.mts`, and `.cts` files (source and test files)

## Type definition files

TypeScript is also able to understand Types on libraries written in pure JavaScript this is done through types definition files.

Some packages come with their own definitions, for the others there are two ways to consume these files.

1.  Use `@types` scoped packages from NPM
1.  Create your own `*.d.ts` file

[Read more](TypeScript_Typings.md)

## Getting Started

We have a user guide to get started with TypeScript in your project

[Follow the guide](Getting_Started_with_TypeScript.md)

### `crafty jsLint`

Leveraging ESLint, we add configuration to lint TypeScript.
You can use that using the same command as for JavaScript.

```bash
crafty jsLint src/**
```

If you pass the `--fix` flag it will fix all the errors it can and write them
directly to the source file.
