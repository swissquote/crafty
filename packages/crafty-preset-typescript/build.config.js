
module.exports = [
  {
    name: "typescript-packages",
    externals: {
      // Dependencies of this package
      "@babel/core": "@babel/core",
      "@babel/code-frame": "@babel/code-frame",
      "@babel/helper-module-imports": "@babel/helper-module-imports",
      typescript: "typescript",

      // Provided by other Crafty packages
      webpack: "@swissquote/crafty-runner-webpack/packages/webpack",
    }
  }
];
