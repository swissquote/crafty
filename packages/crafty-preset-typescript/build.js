#!/usr/bin/env node

const compile = require("../../utils/compile.js");

const externals = {
  "@babel/core": "@babel/core",
  "@babel/code-frame": "@babel/code-frame",
  "@babel/helper-module-imports": "@babel/helper-module-imports",
  typescript: "typescript",
  webpack: "webpack",
  
};

async function main() {
  await compile("./src/typescript-packages.js", "dist/compiled/typescript-packages.js", {
    externals
  });
}

main().then(
  () => {
    console.log("Success");
  },
  e => {
    console.error("Failed", e);
    process.exit(1);
  }
);
