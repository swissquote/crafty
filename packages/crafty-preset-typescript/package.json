{
  "name": "@swissquote/crafty-preset-typescript",
  "version": "1.12.0",
  "license": "Apache-2.0",
  "repository": "https://github.com/swissquote/crafty.git",
  "publishConfig": {
    "access": "public"
  },
  "main": "src/index.js",
  "dependencies": {
    "@babel/core": "7.11.6",
    "@babel/runtime": "7.11.2",
    "@rollup/plugin-babel": "5.2.1",
    "@swissquote/babel-preset-swissquote": "1.12.0",
    "@swissquote/crafty-preset-eslint": "1.12.0",
    "babel-loader": "8.1.0",
    "find-up": "^5.0.0",
    "fork-ts-checker-webpack-plugin": "5.2.0",
    "gulp-babel": "8.0.0",
    "gulp-concat": "2.6.1",
    "gulp-eslint7": "0.3.0",
    "gulp-newer": "1.4.0",
    "gulp-sourcemaps": "2.6.5",
    "gulp-typescript": "5.0.1",
    "rollup-plugin-typescript2": "0.27.2",
    "tmp": "0.2.1",
    "ts-jest": "26.4.0",
    "ts-loader": "8.0.4",
    "ts-pnp": "^1.1.2",
    "typescript": "4.0.3"
  },
  "engines": {
    "node": ">=10.0.0"
  }
}
