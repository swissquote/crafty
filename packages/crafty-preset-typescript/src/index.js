const path = require("path");

const findUp = require("find-up");

const createTask = require("./gulp");

const MODULES = path.join(__dirname, "..", "node_modules");

function absolutePath(item) {
  return path.isAbsolute(item) ? item : path.join(process.cwd(), item);
}

module.exports = {
  presets: [require.resolve("@swissquote/crafty-preset-eslint")],
  defaultConfig() {
    return {
      bundleTypes: { js: "js" }
    };
  },
  bundleCreator(crafty) {
    const configurators = { js: {} };

    if (
      crafty.config.loadedPresets.some(
        preset => preset.presetName === "@swissquote/crafty-runner-gulp"
      )
    ) {
      configurators.js["gulp/typescript"] = (
        craftyAgain,
        bundle,
        gulp,
        StreamHandler
      ) => {
        gulp.task(
          bundle.taskName,
          createTask(craftyAgain, bundle, StreamHandler)
        );
        craftyAgain.watcher.add(bundle.watch || bundle.source, bundle.taskName);
      };
    }

    return configurators;
  },
  rollup(crafty, bundle, rollupConfig) {
    rollupConfig.input.plugins.typescript = {
      plugin: require("rollup-plugin-typescript2"),
      weight: 20,
      options: {
        tsconfigOverride: {
          compilerOptions: {
            // Transpile to esnext so that Babel can apply all its magic
            target: "ESNext",
            // Preserve JSX so babel can optimize it, or add development/debug information
            jsx: "Preserve"
          }
        }
      }
    };

    const babelConfigurator = require("@swissquote/babel-preset-swissquote/configurator-rollup");
    const { hasRuntime, options } = babelConfigurator(crafty, bundle);
    options.extensions = [".ts", ".tsx"];

    if (hasRuntime) {
      rollupConfig.input.external.push(/@babel\/runtime/);
    }

    rollupConfig.input.plugins.babelTypeScript = {
      plugin: require("@rollup/plugin-babel"),
      weight: 30,
      options
    };
  },
  eslint(config, eslint) {
    // This configuration is read by the webpack and rollup plugins
    // The rest of the configuration is handled by `eslint-plugin-swissquote`
    eslint.extensions.push("ts");
    eslint.extensions.push("tsx");

    return eslint;
  },
  jest(crafty, options) {
    options.moduleDirectories.push(MODULES);
    options.transform["^.+\\.tsx?$"] = require.resolve("ts-jest");

    options.moduleFileExtensions.push("ts");
    options.moduleFileExtensions.push("tsx");
  },
  webpack(crafty, bundle, chain) {
    const configFile = findUp.sync("tsconfig.json", { cwd: process.cwd() });

    if (!configFile) {
      crafty.log.error(
        `No tsconfig.json found in "${process.cwd()}". Skipping initialization of TypeScript loaders.`
      );
      return;
    }

    chain.resolve.extensions.add(".ts").add(".tsx");
    chain.resolve.modules.add(MODULES);
    chain.resolveLoader.modules.add(MODULES);

    // TypeScript
    const tsRule = chain.module.rule("ts");
    tsRule.test(/\.tsx?$/);
    tsRule.exclude.add(/(node_modules|bower_components)/);

    const babelConfigurator = require("@swissquote/babel-preset-swissquote/configurator-webpack");
    const babelOptions = babelConfigurator(crafty, bundle);

    // EcmaScript 2015+
    tsRule
      .use("babel")
      .loader(require.resolve("babel-loader"))
      .options(babelOptions);

    const tsOptions = {
      // https://webpack.js.org/guides/build-performance/#typescript-loader
      experimentalWatchApi: true,
      compilerOptions: {
        // Transpile to esnext so that Babel can apply all its magic
        target: "ESNext",
        // Preserve JSX so babel can optimize it, or add development/debug information
        jsx: "Preserve"
      }
    };

    // Get the current configuration to know what configuration options we have to set
    const compiler = require("typescript");
    const currentConfig = compiler.readConfigFile(
      configFile,
      compiler.sys.readFile
    );

    const hasDeclarations =
      currentConfig.config &&
      currentConfig.config.compilerOptions &&
      currentConfig.config.compilerOptions.declaration;

    if (
      hasDeclarations &&
      !currentConfig.config.compilerOptions.declarationDir
    ) {
      // Write declaration files in the destination folder
      // We set the value this way to respect backwards compatibility,
      // Ideally, the value should be without the `/js` at the end
      tsOptions.compilerOptions.declarationDir = absolutePath(
        `${crafty.config.destination_js +
          (bundle.directory ? `/${bundle.directory}` : "")}/js`
      );
    }

    if (crafty.isPNP) {
      tsOptions.resolveModuleName = require("ts-pnp").resolveModuleName;
    }

    // Fork-ts-webpack checker is enabled only if we don't have declarations enabled in our tsconfig.json
    // https://github.com/Realytics/fork-ts-checker-webpack-plugin/issues/49
    if (!hasDeclarations) {
      tsOptions.transpileOnly = true;

      const forkCheckerOptions = {
        typescript: {
          typescriptPath: require.resolve("typescript"),
          configOverwrite: {
            compilerOptions: tsOptions.compilerOptions
          }
        }
      };

      chain
        .plugin("fork-ts-checker")
        .use(require.resolve("fork-ts-checker-webpack-plugin"), [
          forkCheckerOptions
        ]);
    }

    tsRule
      .use("ts-loader")
      .loader(require.resolve("ts-loader"))
      .options(tsOptions);
  }
};
