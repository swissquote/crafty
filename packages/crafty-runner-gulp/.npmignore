build.config.js
src
dist/*/*-stats.json
reports
