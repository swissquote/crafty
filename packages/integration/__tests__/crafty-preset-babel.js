/* global describe, it, expect, jest */

const fs = require("fs");
const path = require("path");

const rimraf = require("rimraf");
const configuration = require("@swissquote/crafty/src/configuration");
const getCommands = require("@swissquote/crafty/src/commands/index");

const testUtils = require("../utils");

const getCrafty = configuration.getCrafty;

// node-forge 0.6.33 doesn't work with jest.
// but selfsigned is fixed on this version
// as we don't use it for now, we can simply mock it
// https://github.com/jfromaniello/selfsigned/issues/16
jest.mock("node-forge");

it("Loads crafty-preset-babel and does not register webpack tasks", () => {
  const crafty = getCrafty(["@swissquote/crafty-preset-babel"], {});

  const loadedPresets = [
    require("@swissquote/crafty-preset-babel"),
    { presetName: "crafty.config.js" }
  ];

  expect(crafty.config.loadedPresets).toEqual(loadedPresets);

  const commands = getCommands(crafty);
  expect(Object.keys(commands)).toEqual([
    "jsLint",
    "help",
    "run",
    "watch",
    "test"
  ]);

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([]);
});

it("Loads crafty-preset-babel, crafty-runner-webpack and registers webpack task", () => {
  const config = { js: { myBundle: { source: "css/style.scss" } } };
  const crafty = getCrafty(
    ["@swissquote/crafty-preset-babel", "@swissquote/crafty-runner-webpack"],
    config
  );

  const loadedPresets = [
    require("@swissquote/crafty-preset-babel"),
    require("@swissquote/crafty-runner-webpack"),
    Object.assign({ presetName: "crafty.config.js" }, config)
  ];

  expect(crafty.config.loadedPresets).toEqual(loadedPresets);

  const commands = getCommands(crafty);
  expect(Object.keys(commands)).toEqual([
    "jsLint",
    "help",
    "run",
    "watch",
    "test"
  ]);

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([
    "js_myBundle",
    "js",
    "default"
  ]);
});

it("Fails on double runner with incorrect bundle assignment", () => {
  const config = { js: { myBundle: { source: "css/style.scss" } } };
  const crafty = getCrafty(
    [
      "@swissquote/crafty-preset-babel",
      "@swissquote/crafty-runner-gulp",
      "@swissquote/crafty-runner-webpack"
    ],
    config
  );

  const loadedPresets = [
    require("@swissquote/crafty-preset-babel"),
    require("@swissquote/crafty-runner-gulp"),
    require("@swissquote/crafty-runner-webpack"),
    Object.assign({ presetName: "crafty.config.js" }, config)
  ];

  expect(crafty.config.loadedPresets).toEqual(loadedPresets);

  expect(() => crafty.createTasks()).toThrow(
    "You have multiple runners, please specify a runner for 'myBundle'. Available runners are ['gulp/babel', 'webpack']."
  );
});

it("Fails on double runner with imprecise bundle assignment", () => {
  const config = {
    js: { myBundle: { runner: "gulp", source: "css/style.scss" } }
  };
  const crafty = getCrafty(
    [
      "@swissquote/crafty-preset-babel",
      "@swissquote/crafty-preset-typescript",
      "@swissquote/crafty-runner-gulp"
    ],
    config
  );

  const loadedPresets = [
    require("@swissquote/crafty-preset-babel"),
    require("@swissquote/crafty-preset-typescript"),
    require("@swissquote/crafty-runner-gulp"),
    Object.assign({ presetName: "crafty.config.js" }, config)
  ];

  expect(crafty.config.loadedPresets).toEqual(loadedPresets);

  expect(() => crafty.createTasks()).toThrow(
    "More than one valid runner exists for 'myBundle'. Has to be one of ['gulp/babel', 'gulp/typescript']."
  );
});

it("Fails on non-existing runners", () => {
  const config = {
    js: { myBundle: { runner: "someRunner", source: "css/style.scss" } }
  };
  const crafty = getCrafty(
    [
      "@swissquote/crafty-preset-babel",
      "@swissquote/crafty-preset-typescript",
      "@swissquote/crafty-runner-gulp"
    ],
    config
  );

  const loadedPresets = [
    require("@swissquote/crafty-preset-babel"),
    require("@swissquote/crafty-preset-typescript"),
    require("@swissquote/crafty-runner-gulp"),
    Object.assign({ presetName: "crafty.config.js" }, config)
  ];

  expect(crafty.config.loadedPresets).toEqual(loadedPresets);

  expect(() => crafty.createTasks()).toThrow(
    "Invalid runner 'someRunner' for 'myBundle'. Has to be one of ['gulp/babel', 'gulp/typescript']."
  );
});

it("Assigns bundle only once when runner is specified", () => {
  const config = {
    js: { myBundle: { runner: "webpack", source: "css/style.scss" } }
  };
  const crafty = getCrafty(
    [
      "@swissquote/crafty-preset-babel",
      "@swissquote/crafty-runner-gulp",
      "@swissquote/crafty-runner-webpack"
    ],
    config
  );

  const loadedPresets = [
    require("@swissquote/crafty-preset-babel"),
    require("@swissquote/crafty-runner-gulp"),
    require("@swissquote/crafty-runner-webpack"),
    Object.assign({ presetName: "crafty.config.js" }, config)
  ];

  expect(crafty.config.loadedPresets).toEqual(loadedPresets);

  const commands = getCommands(crafty);
  expect(Object.keys(commands)).toEqual([
    "jsLint",
    "help",
    "run",
    "watch",
    "test"
  ]);

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([
    "js_myBundle",
    "js",
    "default"
  ]);
});

it("Lints JavaScript using command", () => {
  process.chdir(path.join(__dirname, "../fixtures/crafty-preset-babel/lints"));
  rimraf.sync("dist");

  const result = testUtils.run(["jsLint", "js/**/*.js"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints JavaScript using command, ignore crafty.config.js", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-babel/lints-ignore-config")
  );
  rimraf.sync("dist");

  const result = testUtils.run([
    "--preset",
    "@swissquote/crafty-preset-babel",
    "--ignore-crafty-config",
    "jsLint",
    "crafty.config.js"
  ]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints JavaScript using command, legacy", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-babel/lints-es5")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["jsLint", "js/**/*.js", "--preset", "legacy"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints JavaScript using command, recommended preset", () => {
  process.chdir(path.join(__dirname, "../fixtures/crafty-preset-babel/lints"));
  rimraf.sync("dist");

  const result = testUtils.run([
    "jsLint",
    "js/**/*.js",
    "--preset",
    "recommended"
  ]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints JavaScript using command, explicit configuration", () => {
  process.chdir(path.join(__dirname, "../fixtures/crafty-preset-babel/lints"));
  rimraf.sync("dist");

  const result = testUtils.run([
    "jsLint",
    "js/**/*.js",
    "--config",
    "eslintOverride.json"
  ]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});
