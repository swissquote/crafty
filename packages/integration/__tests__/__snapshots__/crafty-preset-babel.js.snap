// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates IDE Helper 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
Written prettier.config.js
Written .eslintrc.js
Written .gitignore
",
}
`;

exports[`Generates IDE Helper 2`] = `
"// This configuration was generated by Crafty
// This file is generated to improve IDE Integration
// You don't need to commit this file, nor need it to run \`crafty build\`

// Fix module resolution
require(\\"__PATH__/packages/crafty-preset-eslint/src/patchModuleResolver.js\\");

module.exports = {
    \\"plugins\\": [
        \\"@swissquote/swissquote\\"
    ],
    \\"extends\\": [
        \\"plugin:@swissquote/swissquote/format\\",
        \\"plugin:@swissquote/swissquote/recommended\\"
    ],
    \\"rules\\": {
        \\"no-var\\": 0
    },
    \\"settings\\": {}
};
"
`;

exports[`Generates IDE Helper 3`] = `
"// This configuration was generated by Crafty
  // This file is generated to improve IDE Integration
  // You don't need to commit this file

  module.exports = {};
"
`;

exports[`Generates IDE Helper 4`] = `
"/prettier.config.js
/.eslintrc.js
"
`;

exports[`Lints JavaScript using command 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/Component.js
   4:1   error  Delete \`⏎\`                                 @swissquote/swissquote/prettier/prettier
   6:20  error  Replace \`hey:·true⏎···\` with \`·hey:·true\`  @swissquote/swissquote/prettier/prettier
  10:11  error  Delete \`⏎·\`                                @swissquote/swissquote/prettier/prettier
  12:11  error  Delete \`·\`                                 @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/script.js
   3:23  error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
   5:10  error    'test' is defined but never used                 no-unused-vars
   5:14  error    Replace \`·(one,\` with \`(one,·\`                   @swissquote/swissquote/prettier/prettier
   6:1   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                     no-console
  14:9   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  15:5   warning  Unexpected console statement                     no-console
  15:17  error    Replace \`··\\"some·method\\"·\` with \`\\"some·method\\"\`  @swissquote/swissquote/prettier/prettier
  18:11  error    Replace \`·a,·b)⏎·\` with \`a,·b)\`                  @swissquote/swissquote/prettier/prettier
  20:14  error    Expected '===' and instead saw '=='              eqeqeq

✖ 14 problems (12 errors, 2 warnings)
  10 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;

exports[`Lints JavaScript using command, explicit configuration 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/Component.js
   4:1   error  Delete \`⏎\`                                 @swissquote/swissquote/prettier/prettier
   6:20  error  Replace \`hey:·true⏎···\` with \`·hey:·true\`  @swissquote/swissquote/prettier/prettier
  10:11  error  Delete \`⏎·\`                                @swissquote/swissquote/prettier/prettier
  12:11  error  Delete \`·\`                                 @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/script.js
   3:23  error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
   5:10  error    'test' is defined but never used                 no-unused-vars
   5:14  error    Replace \`·(one,\` with \`(one,·\`                   @swissquote/swissquote/prettier/prettier
   6:1   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                     no-console
  14:9   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  15:5   warning  Unexpected console statement                     no-console
  15:17  error    Replace \`··\\"some·method\\"·\` with \`\\"some·method\\"\`  @swissquote/swissquote/prettier/prettier
  18:11  error    Replace \`·a,·b)⏎·\` with \`a,·b)\`                  @swissquote/swissquote/prettier/prettier
  20:14  error    Expected '===' and instead saw '=='              eqeqeq

✖ 14 problems (12 errors, 2 warnings)
  10 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;

exports[`Lints JavaScript using command, format preset 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/Component.js
   4:1   error  Delete \`⏎\`                                 @swissquote/swissquote/prettier/prettier
   6:20  error  Replace \`hey:·true⏎···\` with \`·hey:·true\`  @swissquote/swissquote/prettier/prettier
  10:11  error  Delete \`⏎·\`                                @swissquote/swissquote/prettier/prettier
  12:11  error  Delete \`·\`                                 @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/script.js
   3:23  error  Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
   5:14  error  Replace \`·(one,\` with \`(one,·\`                   @swissquote/swissquote/prettier/prettier
   6:1   error  Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  14:9   error  Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  15:17  error  Replace \`··\\"some·method\\"·\` with \`\\"some·method\\"\`  @swissquote/swissquote/prettier/prettier
  18:11  error  Replace \`·a,·b)⏎·\` with \`a,·b)\`                  @swissquote/swissquote/prettier/prettier

✖ 10 problems (10 errors, 0 warnings)
  10 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;

exports[`Lints JavaScript using command, ignore crafty.config.js 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints-ignore-config/crafty.config.js
  1:1  error  'module' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)

",
}
`;

exports[`Lints JavaScript using command, legacy 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints-es5/js/script.js
  1:1   error  Use the function form of 'use strict'  strict
  1:14  error  Replace \`·(one,\` with \`(one,·\`         @swissquote/swissquote/prettier/prettier
  2:1   error  Delete \`·\`                             @swissquote/swissquote/prettier/prettier
  5:1   error  Use the function form of 'use strict'  strict
  5:23  error  Replace \`⏎\` with \`·\`                   @swissquote/swissquote/prettier/prettier
  7:12  error  Expected '===' and instead saw '=='    eqeqeq

✖ 6 problems (6 errors, 0 warnings)
  3 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;

exports[`Lints JavaScript using command, recommended preset 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/Component.js
   4:1   error  Delete \`⏎\`                                 @swissquote/swissquote/prettier/prettier
   6:20  error  Replace \`hey:·true⏎···\` with \`·hey:·true\`  @swissquote/swissquote/prettier/prettier
  10:11  error  Delete \`⏎·\`                                @swissquote/swissquote/prettier/prettier
  12:11  error  Delete \`·\`                                 @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-babel/lints/js/script.js
   3:23  error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
   5:10  error    'test' is defined but never used                 no-unused-vars
   5:14  error    Replace \`·(one,\` with \`(one,·\`                   @swissquote/swissquote/prettier/prettier
   6:1   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                     no-console
  14:9   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  15:5   warning  Unexpected console statement                     no-console
  15:17  error    Replace \`··\\"some·method\\"·\` with \`\\"some·method\\"\`  @swissquote/swissquote/prettier/prettier
  18:11  error    Replace \`·a,·b)⏎·\` with \`a,·b)\`                  @swissquote/swissquote/prettier/prettier
  20:14  error    Expected '===' and instead saw '=='              eqeqeq

✖ 14 problems (12 errors, 2 warnings)
  10 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;
